src/main/java/pages/PageObject.java

```diff
@@ -1,15 +1,39 @@
 package pages;
 
+import hooks.Hooks;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.PageFactory;
 
 public abstract class PageObject {
 
-    protected WebDriver driver;
+    private WebDriver driver = Hooks.getDriver();
+    private HomePage homePage;
+    private ProductPage productPage;
+    private ShoppingCart shoppingCart;
+    private ProductDetailsPage productDetailsPage;
 
-    PageObject(WebDriver driver) {
-        this.driver = driver;
+    public PageObject() {
         PageFactory.initElements(driver, this);
     }
```

Comment:
revert this change, Page object constructor must take as input parameter the webdriver